pauliopt/pauli/clifford_gates.py

```diff
@@ -0,0 +1,274 @@
+from enum import Enum
+from typing import List, Collection
+import numpy as np
+from .utils import _pauli_to_string, X, Y, Z, I, Pauli
+
+from .pauli_gadget import PauliGadget
+
+
+class CLiffordType(Enum):
+    CX = "cx"
+    CY = "cy"
+    CZ = "cz"
+    H = "h"
+    S = "s"
+    V = "v"
+
+
+class CliffordGate:
+    def __init__(self, c_type):
+        self.c_type = c_type
+
+    def propagate_pauli(self, gadget: PauliGadget):
+        raise Exception(f"This method is not implemented on object: {self}")
+
+    def to_qiskit(self):
+        raise Exception(f"This method is not implemented on object: {self}")
+
+    @property
+    def num_qubits(self):
+        raise Exception(f"This property is not implemented on object: {self}")
+
+    @staticmethod
+    def generate_random(num_qubits):
```

Comment:
`generate_random` can be a utility function that takes a gate or a set of gates, and returns a random circuit. I see some repeated code across the concrete classes here.